modules/instrument_manager/jsx/instrumentManagerIndex.js

```diff
@@ -0,0 +1,174 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+
+import {Tabs, TabPane} from 'Tabs';
+import Loader from 'Loader';
+import FilterableDataTable from 'FilterableDataTable';
+
+import InstrumentUploadForm from './uploadForm';
+
+class InstrumentManagerIndex extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {},
+      error: false,
+      isLoaded: false,
+    };
+
+   this.fetchData = this.fetchData.bind(this);
+    this.formatColumn = this.formatColumn.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData()
+      .then(() => this.setState({isLoaded: true}));
+  }
+
+  /**
+   * Retrieve data from the provided URL and save it in state
+   * Additionally add hiddenHeaders to global loris variable
+   * for easy access by columnFormatter.
+   *
+   * @return {object}
+   */
+  fetchData() {
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({data}))
+      .catch((error) => {
+        this.setState({error: true});
+        console.error(error);
+      });
+  }
+
+  /**
+   * Modify behaviour of specified column cells in the Data Table component
+   *
+   * @param {string} column - column name
+   * @param {string} cell - cell content
+   * @param {object} row - row content indexed by column
+   *
+   * @return {*} a formated table cell for a given column
+   */
+  formatColumn(column, cell, row) {
+    // Set class to 'bg-danger' if file is hidden.
+    return (
+      <td>{cell}</td>
+    );
+  }
+
+  render() {
+    // If error occurs, return a message.
+    // XXX: Replace this with a UI component for 500 errors.
+    if (this.state.error) {
+      return <h3>An error occured while loading the page.</h3>;
+    }
+
+    // Waiting for async data to load
+    if (!this.state.isLoaded) {
+      return <Loader/>;
+    }
+
+    const fields = [
+      {label: 'Instrument', show: true, filter: {
+        name: 'instrument',
+        type: 'text',
+      }},
+      {label: 'Instrument Type', show: true, filter: {
+        name: 'instrument_type',
+        type: 'select',
+        options: {
+          'Instrument Builder': 'Instrument Builder',
+          'PHP': 'PHP',
+          'Missing': 'Missing',
+        },
+      }},
+      {label: 'Table Installed', show: true, filter: {
+        name: 'table_installed',
+        type: 'select',
+        options: {
+          'Exists': 'Exists',
+          'Missing': 'Missing',
+        },
+      }},
+      {label: 'Table Valid', show: true, filter: {
+        name: 'table_valid',
+        type: 'text',
+      }},
+      {label: 'Pages Valid', show: true, filter: {
+        name: 'pages_valid',
+        type: 'text',
+      }},
+    ];
+
+    const tabs = [
+      {id: 'browse', label: 'Browse'},
+      {id: 'upload', label: 'Upload'},
+    ];
+
+    const feedback = () => {
+      if (!this.state.data.caninstall) {
```

Comment:
Perhaps `canInstall` shuold be passed in a permission object? It just seems to odd to have it as part of 'data'. This may require some restructuring of how we handle permissions on the front end. Is there a way to access this information via `this.props.hasPermission`?

Reply:
I don't think so. Whether or not they can install is a function of the filesystem permissions/setup, not the user's permission.

Reply:
I partly agree with you. Yes, it is odd the it is part of the data but it is not a permission either. This tells if the project/instrument and project/table_sql directory can be written into.

Reply:
@driusan @xlecours Hmm, okay. Yeah, that makes. I guess as we are making standards for the objects we are passing to the front end (e.g. `data`, `fieldOptions`, etc.), we may want to group it into a `configuration` object? Rather than have stray configuration booleans in the `data` object. Not necessary for this PR, just something to think about.